### PR TITLE
perf: remove channel as mailbox and use queue implementation

### PR DESCRIPTION
**Pull request by Tochemey**
- Remove channel as mailbox because of performance issue
- Replace mailbox with internal.Queue implementation

### CODE REVIEW

File: internal/queue/queue.go
Comment (halturin) [inline internal/queue/queue.go:1]: You can improve performance by using MPSC queue (lock-free). Take inspiration from https://github.com/ergo-services/ergo/blob/v300/lib/mpsc.go

Comment (Tochemey) [reply internal/queue/queue.go:1]: @halturin Thanks for the suggestion. I will give it a try.

Comment (Tochemey) [reply internal/queue/queue.go:1]: @halturin I love the fact ergo is the go erlang. Nice library by the way :)

Comment (sdil) [reply internal/queue/queue.go:1]: i just realize the word 😆 erlang + go = ergo

**Pull request merged**